### DISCUSSION TITLE
[boost] Build error - boost fails to install in manifest mode with version override

### DISCUSSION THREAD

**Discussion by hros** [microsoft/vcpkg#28185] [Q&A]
I am using vcpkg manifest to select a specific version of boost. The `vcpkg.json` contains:
```json
  "dependencies": [
    ....... ,
    "boost",
  ],
  "overrides": [
    {
      "name": "boost",
      "version": "1.65"
    }
  ],
```

When I install the dependencies (either with `vcpkg install` or with the `cmake` command) the installation fails.
Note that I am running in Ubuntu-20.04 in wsl2. I'd appreciate advice how to resolve the issue

Package: boost[core]:x64-linux -> 1.65

**Host Environment**

- Host: x64-linux
- Compiler: GNU 9.4.0
-    vcpkg-tool version: 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba
    vcpkg-scripts version: 163fe7bd3 2022-12-02 (3 days ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

The output of `build/vcpkg-manifest-install.log`:

> Installing 5/71 boost:x64-linux...
Building boost[core]:x64-linux...
-- Installing port from location: /home/user/.local/vcpkg/buildtrees/versioning_/versions/boost/f666638c221fa08912348791f530970beba055b6
CMake Warning at scripts/cmake/vcpkg_common_functions.cmake:3 (message):
  vcpkg_common_functions has been removed and all values are automatically
  provided in all portfile.cmake invocations.  Please remove
  `include(vcpkg_common_functions)`.
Call Stack (most recent call first):
  buildtrees/versioning_/versions/boost/f666638c221fa08912348791f530970beba055b6/portfile.cmake:1 (include)
  scripts/ports.cmake:147 (include)
>
>
> -- Using cached boost_1_65_0.7z.
>
> CMake Deprecation Warning at scripts/ports.cmake:77 (message):
>
>   Deprecated form of vcpkg_extract_source_archive used:
> 
>       Please use the `vcpkg_extract_source_archive(<out-var> ARCHIVE <archive>)` form.
> 
> 
>
> Call Stack (most recent call first):
>
> scripts/cmake/vcpkg_extract_source_archive.cmake:18 (z_vcpkg_deprecation_message)
>   buildtrees/versioning_/versions/boost/f666638c221fa08912348791f530970beba055b6/portfile.cmake:14 (vcpkg_extract_source_archive)
>
> scripts/ports.cmake:147 (include)
>
>-- Downloading https://github.com/boostorg/range/commit/e7ebe14707130cda7b72e0ae5e93b17157fdb6a2.diff -> boost-range-has_range_iterator-hotfix_e7ebe14707130cda7b72e0ae5e93b17157fdb6a2-2.diff...
> [DEBUG] To include the environment variables in debug output, pass --debug-env
> [DEBUG] Feature flag 'binarycaching' unset
> [DEBUG] Feature flag 'compilertracking' unset
> [DEBUG] Feature flag 'registries' unset
> [DEBUG] Feature flag 'versions' unset
> [DEBUG] 1000: popen( curl --fail -L https://github.com/boostorg/range/commit/e7ebe14707130cda7b72e0ae5e93b17157fdb6a2.diff --create-dirs --output /home/user/.local/vcpkg/downloads/boost-range-has_range_iterator-hotfix_e7ebe14707130cda7b72e0ae5e93b17157fdb6a2-2.diff.19527.part 2>&1)
[DEBUG] 1000: cmd_execute_and_stream_data() returned 0 after   627525 us
error: Failed to download from mirror set
error: File does not have the expected hash:
             url : [ https://github.com/boostorg/range/commit/e7ebe14707130cda7b72e0ae5e93b17157fdb6a2.diff ]
       File path : [ /home/user/.local/vcpkg/downloads/boost-range-has_range_iterator-hotfix_e7ebe14707130cda7b72e0ae5e93b17157fdb6a2-2.diff.19527.part ]
   Expected hash : [ 88392fcf092d07ee2de66d937634688eef1d7491a4e2fb0fce064b4e631813a0814455a329f9eed8b968ff205883751d51a86edecc8e355790fc8a35d6742483 ]
     Actual hash : [ f931ac52dfa95063c79c746077096f76178f02480f656779d3c38d978cbc9b0115ddf959e8a4c953cb807c06739c59eb0f14b9a9af0effe742d299af61b9392c ]
>
>
> [DEBUG] /mnt/vss/_work/1/s/src/vcpkg/base/downloads.cpp(774):
> [DEBUG] Time in subprocesses: 627525 us
> [DEBUG] Time in parsing JSON: 3 us
> [DEBUG] Time in JSON reader: 0 us
> [DEBUG] Time in filesystem: 14 us
> [DEBUG] Time in loading ports: 0 us
> [DEBUG] Exiting after 627.9 ms (627870 us)
>
> CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
>
>       Failed to download file with error: 1
>       If you use a proxy, please check your proxy setting. Possible causes are:
> 
>       1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
>          to `https://address:port`. This is not correct, because `https://` prefix
>          claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
>          , etc..) is an HTTP proxy. Try setting `http://address:port` to both
>          HTTP_PROXY and HTTPS_PROXY instead.
> 
>       2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
>          will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
>          Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
>          Change `http=address:port;https=address:port` to `address`, and fill the port number.
> 
>       3. Your proxy's remote server is out of service.
> 
>       In future vcpkg releases, if you are using Windows, you no longer need to set
>       HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
>       Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
>       and (https://github.com/microsoft/vcpkg-tool/pull/77)
> 
>       Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_download_distfile.cmake:273 (z_vcpkg_download_distfile_show_proxy_and_fail)
>   buildtrees/versioning_/versions/boost/f666638c221fa08912348791f530970beba055b6/portfile.cmake:17 (vcpkg_download_distfile)
>   scripts/ports.cmake:147 (include)
> 
> 
> error: building boost:x64-linux failed with: BUILD_FAILED
> error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
> Then check for known issues at:
>     https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+boost
> You can submit a new issue at:
>     https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[boost]+Build+error
> Include '[boost] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
>     vcpkg-tool version: 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba
>     vcpkg-scripts version: 163fe7bd3 2022-12-02 (3 days ago)

**Comment by JackBoosY** [comment]
HASH check failure.
BTW, the earliest boost version is 1.66 in vcpkg.

**Reply by dg0yt** [reply]
There is 1.65 in vcpkg. `versions/b-/boost.json` starts with 1.60.

**Comment by hros** [comment]
As you can see from the log, version 1.65 is being downloaded from boost mirrors
Since 1.65 is not supported by vcpkg, it should not be downloaded and built and an appropriate error message should be displayed

**Reply by JackBoosY** [reply]
So, I think you should use overlay-port and modify the hash value since we can't fix it in the master branch.
